pages/scenarios/[runId].vue

```diff
@@ -120,56 +121,82 @@ import type { Parameter } from "~/types/parameterTypes";
+
+    const rawValIsNumberString = Number.parseInt(rawVal).toString() === rawVal;
+    if (rawValIsNumberString) {
+      // TODO: Localize number formatting.
+      return new Intl.NumberFormat().format(Number.parseInt(rawVal));
+    }
     return param.options ? param.options.find(({ id }) => id === rawVal)!.label : rawVal;
   }
 };
 
+// Use useAsyncData to store the time once, during server-side rendering: avoids client render re-writing value.
+const { data: timeOfFirstStatusPoll } = await useAsyncData<number>("timeOfFirstStatusPoll", async () => {
+  return new Promise<number>((resolve) => {
+    resolve(new Date().getTime());
+  });
+});
+
 // Eagerly try to load the status and results, in case they are already available and can be used during server-side rendering.
 await appStore.loadScenarioStatus();
 if (appStore.currentScenario.status.data?.runSuccess) {
   appStore.loadScenarioResult();
 }
 
-let statusInterval: NodeJS.Timeout;
-const loadScenarioStatus = () => {
-  appStore.loadScenarioStatus().then(() => {
-    if (appStore.currentScenario.status.data?.runSuccess) {
-      clearInterval(statusInterval);
-      jobTakingLongTime.value = false;
-      appStore.loadScenarioResult();
-    }
-  });
+watch(() => appStore.currentScenario.status.data?.runSuccess, (runSuccess) => {
+  if (runSuccess) {
+    appStore.loadScenarioResult();
+  }
+});
+
+watch(() => appStore.currentScenario.status.data?.done, (done) => {
+  if (done) {
+    clearInterval(statusInterval);
+    jobSlow.value = false;
+    jobReallySlow.value = false;
+  }
+});
+
+const pollForStatusEveryNSeconds = (seconds: number) => {
+  statusInterval = setInterval(() => {
+    if (timeOfFirstStatusPoll.value) {
+      secondsSinceFirstStatusPoll.value = ((new Date().getTime() - timeOfFirstStatusPoll.value) / 1000).toFixed(0);
+    };
+    appStore.loadScenarioStatus();
+  }, seconds * 1000);
 };
 
 onMounted(() => {
   appStore.globe.interactive = false;
 
   if (!appStore.currentScenario.status.data?.done && appStore.currentScenario.runId) {
-    statusInterval = setInterval(loadScenarioStatus, 200); // Poll for status every N ms
+    pollForStatusEveryNSeconds(0.2);
     setTimeout(() => {
-      // If the job isn't completed within five seconds, give user the information about the run status.
+      // If the job isn't completed within M seconds, give user the information about the run status.
       if (appStore.currentScenario.status.data?.runStatus !== runStatus.Complete) {
-        jobTakingLongTime.value = true;
+        jobSlow.value = true;
       }
     }, 5000);
+    // Some runs take an especially long time, e.g. Singapore + Omicron.
     setTimeout(() => {
-      // If the job isn't completed within 10 seconds, terminate polling for status.
       if (!appStore.currentScenario.status.data?.done) {
-        jobTakingLongTime.value = false;
-        stoppedPolling.value = true;
+        jobSlow.value = false;
+        jobReallySlow.value = true;
```

Comment:
Logic might be a little simpler if you just let slow keep being true when reallySlow is also true